cmd/run.go

```diff
@@ -0,0 +1,137 @@
+package main
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"github.com/SigNoz/signoz-otel-collector/constants"
+	"github.com/SigNoz/signoz-otel-collector/pkg/cache"
+	"github.com/SigNoz/signoz-otel-collector/pkg/env"
+	"github.com/SigNoz/signoz-otel-collector/pkg/log"
+	"github.com/SigNoz/signoz-otel-collector/pkg/storage"
+	"github.com/SigNoz/signoz-otel-collector/service"
+	"github.com/SigNoz/signoz-otel-collector/signozcol"
+	"github.com/spf13/cobra"
+	"go.uber.org/zap"
+)
+
+func registerRun(app *cobra.Command) {
+	var collectorConfig collectorConfig
+	var storageConfig storageConfig
+	var cacheConfig cacheConfig
+
+	cmd := &cobra.Command{
+		Use:   "run",
+		Short: "Run the otel collector",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// Create the logger by taking the log-level from the root command
+			logger := log.NewZapLogger(app.PersistentFlags().Lookup("log-level").Value.String())
+			defer func() {
+				_ = logger.Flush()
+			}()
+
+			return runCollector(
+				cmd.Context(),
+				logger,
+				storageConfig,
+				collectorConfig,
+				cacheConfig,
+			)
+		},
+	}
+
+	collectorConfig.registerFlags(cmd)
+	storageConfig.registerFlags(cmd)
+	cacheConfig.registerFlags(cmd)
```

Comment:
I see these are used in extension/processor components. They should rather be part of the component config section instead of here.

Reply:
How much does it affect if each component opens its own connection? Is that a problem to worry about? As you can it already, it's quite hard to follow how do components work. And once we introduce it into the codebase it's only going to get (ab)used more.

Reply:
In this case alone,
One connection pool for keyauth + One connection pool for limiter + Separate connection pool if other extensions/processors decide to use this per **gateway instance**.

Abuse part should be prevented by effective PR reviews!

Reply:
I think one additional connection is not an issue per gateway instance because the number of gateway instances is relatively small. Once a pattern is introduced to the codebase there is no going back. I do not have a strong opinion but at this point, I see keeping the config and connection separate for components as a better option and revisit when it becomes a problem.